File: dfs.py
import numpy as np

def dfs(i, j):
    global cnt
    cnt += 1
    visited[i][j] = True

    if i > 0 and grid[i - 1][j] == 'Y' and not (visited[i - 1][j]):
        dfs(i - 1, j)
    if i < n - 1 and grid[i + 1][j] == 'Y' and not (visited[i + 1][j]):
        dfs(i + 1, j)
    if j < m - 1 and grid[i][j + 1] == 'Y' and not (visited[i][j + 1]):
        dfs(i, j + 1)
    if j > 0 and grid[i][j - 1] == 'Y' and not (visited[i][j - 1]):
        dfs(i, j - 1)

    if i > 0 and j > 0 and grid[i - 1][j - 1] == 'Y' and not (visited[i - 1][j - 1]):
        dfs(i - 1, j - 1)
    if i < n - 1 and j < m - 1 and grid[i + 1][j + 1] == 'Y' and not (visited[i + 1][j + 1]):
        dfs(i + 1, j + 1)
    if i > 0 and j < m - 1 and grid[i - 1][j + 1] == 'Y' and not (visited[i - 1][j + 1]):
        dfs(i - 1, j + 1)
    if i < n - 1 and j > 0 and grid[i + 1][j - 1] == 'Y' and not (visited[i + 1][j - 1]):
        dfs(i + 1, j - 1)


ans = 0
input = open('C:/Users/sazza/Desktop/422/input.txt', 'r')
lines = input.readlines()
grid = []

for i in lines:
    i = i.strip()
    str = i.split(' ')
    grid.append(str)
#print(grid)

n = len(grid)
m = len(grid[0])
cnt = 0
visited = np.zeros((n, m))
visited = visited > 0

for i in range(n):
    for j in range(m):
        if grid[i][j] == 'Y' and not (visited[i][j]):
            cnt = 0
            dfs(i, j)
            ans = max(ans, cnt)

print(ans)

File: dfs.cpp
#include <bits/stdc++.h>
using namespace std;

const int nmax = 1000+10;
bool visited [nmax][nmax];
char grid[nmax][nmax];

int n,m,cnt;


void dfs (int i, int j){
    cnt++;
    visited[i][j] = true;


    if (i > 0 && grid[i-1][j]=='Y' && !visited[i-1][j])
        dfs(i-1, j);
    if (i < n-1 && grid[i+1][j]=='Y' && !visited[i+1][j])
        dfs(i+1, j);
    if (j < m-1 && grid[i][j+1]=='Y' && !visited[i][j+1])
        dfs(i, j+1);
    if (j > 0 && grid[i][j-1]=='Y' && !visited[i][j-1])
        dfs(i, j-1);

    if (i > 0 && j > 0 && grid[i-1][j-1]=='Y' && !visited[i-1][j-1])
        dfs(i-1, j-1);
    if (i < n-1 && j < m-1 && grid[i+1][j+1]=='Y' && !visited[i+1][j+1])
        dfs(i+1, j+1);
    if (i > 0 && j < m-1 && grid[i-1][j+1]=='Y' && !visited[i-1][j+1])
        dfs(i-1, j+1);
    if (i < n-1 && j > 0 && grid[i+1][j-1]=='Y' && !visited[i+1][j-1])
        dfs(i+1, j-1);




}

int main(){
    cin>>n>>m;
    int ans  = 0;

    for (int i=0;i<n;i++){
        cin>>grid[i];
    }

    for (int i=0;i<n;i++){
        for (int j=0;j<m;j++){
            if (grid[i][j]=='Y' && !visited[i][j]){
                cnt = 0;
                dfs(i, j);
                ans = max(ans,cnt);
            }
        }
    }

    cout<<ans<<endl;
}
